### DISCUSSION TITLE
Updates regarding the implementation of samplers from scratch

### DISCUSSION THREAD

**Discussion by Qazalbash** [Qazalbash/jaxampler#97] [Announcements]
🌟 Hello, community,

We embarked on our journey with Jaxampler by crafting bespoke probabilistic sampling algorithms tailored to our needs. Along the way, we encountered challenges we hadn't foreseen. 🛠️

After careful consideration and thorough exploration of GitHub, we've decided not to reinvent the wheel. Here's why:

1. Other expert teams have developed packages with deeper expertise.
2. These packages receive frequent updates and maintenance.
3. They offer scalability to meet evolving demands.
4. Rigorous testing ensures reliability and performance.

These insights lead us to an exciting decision for the future of GWKokab: we're looking to integrate [NumPyro](https://github.com/pyro-ppl/numpyro). 🚀 Your thoughts and feedback are invaluable as we move forward.

Thank you for your continued support,

Team Jaxampler 🌟

**Comment by IamMuhammadZeeshan** [comment]
I think it makes sense, particularly they wrote various sampler based on JAX. So, better to use them instead of reinventing the wheel.
